src/components/llama-item.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2021 Google Inc. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+import { LitElement, html } from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+
+import { LlamaStorage } from '../storage';
+
+// Whether the Web Share API has been exposed on the browser's Navigator object.
+const kWebShareExposed = 'share' in navigator && 'canShare' in navigator;
+
+@customElement('llama-item')
+export class LlamaItem extends LitElement {
+  @property({ type: String }) id: string = '';
+  @property({ type: Object }) storage?: Promise<LlamaStorage>;
+  @property({ type: String }) src: string = '';
+
+  // No internal behaviour is necessary for the header, so avoid creating an
+  // open shadow root for this component and default to global styles instead.
+  createRenderRoot() { return this; }
+
+  async shareItem() {
+    if (!this.storage) return;
```

Comment:
When would this happen? Would it be down to programmer error or something in the environment?

Reply:
Programmer error - not setting the `@storage` _property_ on the `<llama-item>` element.

In most other environments I'd argue for an exception of sorts, but that's not idiomatic for HTML elements. Similar cases (e.g. `<input>` without `@type`) all silently fall back to some default, e.g. a text box, or an image placeholder when no `@src` has been provided. Unless we can statically assert that this never happens, it seems to me that matching that is the right approach?

Reply:
Fair enough - I was going to suggest an exception, but if that's not the done thing for HTML elements this makes sense. Would it be worth a console warning?

Reply:
Added.